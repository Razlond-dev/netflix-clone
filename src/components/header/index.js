/* eslint-disable react/prop-types */
import React, { useState } from "react";
import { Link } from "react-router-dom";
import {
  Background,
  Logo,
  Container,
  ButtonLink,
  Feature,
  Text,
  TextLink,
  FeatureCallOut,
  Group,
  Picture,
  Profile,
  Dropdown,
  Search,
  SearchIcon,
  SearchInput,
  PlayButton,
} from "./styles/header";

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Group = function HeaderFrame({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <Link to={to}>
      <Logo {...restProps} />
    </Link>
  );
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderText({ children, ...restProps }) {
  return <TextLink {...restProps}>{children}</TextLink>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture src={`/images/users/${src}.png`} />;
};
Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
  return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Search = function HeaderDropdown({
  searchTerm,
  setSearchTerm,
  ...restProps
}) {
  const [searchActive, setSearchActive] = useState(false);

  return (
    <Search {...restProps}>
      <SearchIcon
        onClick={() => setSearchActive((searchActive) => !searchActive)}
      >
        <img src="/images/icons/search.png" alt="Search" />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={({ target }) => setSearchTerm(target.value)}
        placeholder="Search films and series"
        active={searchActive}
      />
    </Search>
  );
};
