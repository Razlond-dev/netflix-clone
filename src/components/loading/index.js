import React from "react";
import { LockBody, Picture, ReleaseBody, Spinner } from "./styles/loading";

export default function Loading({ src, ...restProps }) {
  return (
    <Spinner data-testid="loading">
      <LockBody />
      <Picture src={`/images/users/${src}.png`} data-testid="loading-picture" />
    </Spinner>
  );
}

Loading.RealeaseBody = function LoadingReleaseBody() {
  return <ReleaseBody />;
};
