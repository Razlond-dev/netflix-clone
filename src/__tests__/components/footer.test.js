import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { Footer } from "../../components";

describe("<Footer />", () => {
  it("renders the <Footer /> with populated data", () => {
    const { container, getByText, queryByTestId } = render(
      <Footer>
        <Footer.Title>Questions?</Footer.Title>
        <Footer.Break />
        <Footer.Row>
          <Footer.Column>
            <Footer.Link href="#">FAQ</Footer.Link>
            <Footer.Link href="#">Investor Relations</Footer.Link>
            <Footer.Link href="#">Ways to Watch</Footer.Link>
            <Footer.Link href="#">More information</Footer.Link>
            <Footer.Link href="#">Netflix Originals</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Help</Footer.Link>
            <Footer.Link href="#">Jobs</Footer.Link>
            <Footer.Link href="#">Terms of Use</Footer.Link>
            <Footer.Link href="#">Contact us</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Account</Footer.Link>
            <Footer.Link href="#">Gift Cards </Footer.Link>
            <Footer.Link href="#">Privacy</Footer.Link>
            <Footer.Link href="#">Legal Notices</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Events</Footer.Link>
            <Footer.Link href="#">New Films</Footer.Link>
            <Footer.Link href="#">Top 100</Footer.Link>
            <Footer.Link href="#">Our Products</Footer.Link>
          </Footer.Column>
        </Footer.Row>
        <Footer.Break />
        <Footer.Text>Netflix United Kingdom</Footer.Text>
      </Footer>
    );

    expect(getByText("FAQ")).toBeTruthy();
    expect(getByText("Investor Relations")).toBeTruthy();
    expect(getByText("Netflix Originals")).toBeTruthy();
    expect(getByText("Contact us")).toBeTruthy();
    expect(getByText("Legal Notices")).toBeTruthy();
    expect(getByText("Events")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
