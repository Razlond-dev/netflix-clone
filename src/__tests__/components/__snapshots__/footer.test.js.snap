// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the <Footer /> with populated data 1`] = `
<div
  class="footer__Container-eh6b4z-0 kpqbmQ"
>
  <p
    class="footer__Title-eh6b4z-4 dQKxha"
  >
    Questions?
  </p>
  <p
    class="footer__Break-eh6b4z-6 hHvrrV"
  />
  <div
    class="footer__Row-eh6b4z-2 khVTlG"
  >
    <div
      class="footer__Column-eh6b4z-1 aKanr"
    >
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        FAQ
      </a>
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        Ways to Watch
      </a>
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        More information
      </a>
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        Netflix Originals
      </a>
    </div>
    <div
      class="footer__Column-eh6b4z-1 aKanr"
    >
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        Help
      </a>
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        Jobs
      </a>
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        Terms of Use
      </a>
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        Contact us
      </a>
    </div>
    <div
      class="footer__Column-eh6b4z-1 aKanr"
    >
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        Account
      </a>
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        Gift Cards 
      </a>
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        Privacy
      </a>
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        Legal Notices
      </a>
    </div>
    <div
      class="footer__Column-eh6b4z-1 aKanr"
    >
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        Events
      </a>
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        New Films
      </a>
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        Top 100
      </a>
      <a
        class="footer__Link-eh6b4z-3 kWqRKg"
        href="#"
      >
        Our Products
      </a>
    </div>
  </div>
  <p
    class="footer__Break-eh6b4z-6 hHvrrV"
  />
  <p
    class="footer__Text-eh6b4z-5 hIxdYT"
  >
    Netflix United Kingdom
  </p>
</div>
`;
